okex-client/src/client/mod.rs

```diff
@@ -354,30 +407,44 @@ impl OkexClient {
         let headers = self.get_request_headers(request_path)?;
 
         let response = self
-            .client
+            .rate_limit_client(request_path)
+            .await
             .get(Self::url_for_path(request_path))
             .headers(headers)
             .send()
             .await?;
 
-        let positions_data = Self::extract_response_data::<PositionData>(response).await?;
+        let position_data = Self::extract_response_data::<PositionData>(response).await?;
 
         Ok(PositionId {
-            value: positions_data.pos_id,
+            value: position_data.pos_id,
         })
     }
 
     pub async fn close_positions(
         &self,
-        inst_id: String,
-        pos_side: String,
-        margin_mode: String,
+        inst_id: OkexInstrumentId,
+        pos_side: OkexPositionSide,
+        margin_mode: OkexMarginMode,
         ccy: String,
         auto_cxl: bool,
     ) -> Result<ClosePositionData, OkexClientError> {
+        let instrument = match inst_id {
+            OkexInstrumentId::BtcUsdSwap => "BTC-USD-SWAP".to_string(),
```

Comment:
better to implement `Display` and then call `.to_string()`... goes for all of the enums that have fixed string values.